src/robot/libraries/String.py

```diff
@@ -74,6 +74,21 @@ def convert_to_uppercase(self, string):
         """
         return string.upper()
 
+    def convert_to_titlecase(self, string):
+        """Converts string to titlecase. It's preserve space also.
+
+        Examples:
+        | ${str1} = | Convert To Titlecase | abc |
+        | ${str2} = | Convert To Titlecase | i love robot Framework |
+        | ${str3} = | Convert To Titlecase | i love ROBOT Framework |
+        | Should Be Equal | ${str1} | Abc |
+        | Should Be Equal | ${str2} | I Love Robot Framework |
+        | Should Be Equal | ${str3} | I Love ROBOT Framework |
+
+        New in Robot Framework 3.0.2.
+        """
+        return  ''.join( (c.upper() if i == 0 or string[i-1] == ' ' else c) for i, c in enumerate(string))
```

Comment:
This implementation is pretty cleaver, but it doesn't take into account that there are other characters than the ASCII space that should be taken into account when deciding which characters to capitalize. See how Python's `.title()` handles, for example, the tab character (`\t`) and NO-BREAK-SPACE (`u`xa0`):

```python
>>> u'foo\tbar'.title()
u'Foo\tBar'
>>> u'foo\xa0bar'.title()
u'Foo\xa0Bar'
```

We can discuss about better algorithms in the comments of this PR separately.

Reply:
Why not simply `return string.title()`?
That way Robot Framework keeps in line with Python's implementation and makes this code easier.

Reply:
Because `string.title()` doesn't work well in many cases. For example, `'XML and HTML'.title()` returns `'Xml And Html'` which most people consider wrong.

As I've commented below, this is surprisingly complicated problem and it may also depend on the context how title-casing should work. One possibility would be making the algorithm to use configurable. Or then we can decide not to implement this keyword in the first place and let everyone implement their own context specific keywords.